target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c

```diff
@@ -1542,17 +1541,14 @@ static int fe_probe(struct platform_device *pdev)
 	else
 		soc->reg_table = fe_reg_table;
 
-	fe_base = devm_ioremap_resource(&pdev->dev, res);
-	if (IS_ERR(fe_base)) {
-		err = -EADDRNOTAVAIL;
-		goto err_out;
-	}
+	fe_base = devm_platform_ioremap_resource(pdev, 0);
+	if (IS_ERR(fe_base))
+		return PTR_ERR(fe_base);
 
-	netdev = alloc_etherdev(sizeof(*priv));
+	netdev = devm_alloc_etherdev_mqs(&pdev->dev, sizeof(*priv), 4, 4);
```

Comment:
This old ralink driver doesn't use multiple queue I think?

Reply:
I wonder where I got this 4 from...

@namiltd try

```
--- a/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c
+++ b/target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c
@@ -1545,7 +1545,7 @@ static int fe_probe(struct platform_device *pdev)
        if (IS_ERR(fe_base))
                return PTR_ERR(fe_base);
 
-       netdev = devm_alloc_etherdev_mqs(&pdev->dev, sizeof(*priv), 4, 4);
+       netdev = devm_alloc_etherdev(&pdev->dev, sizeof(*priv));
        if (!netdev) {
                dev_err(&pdev->dev, "alloc_etherdev failed\n");
                return -ENOMEM;

```

Reply:
It works, please prepare PR